File: Angular/medpr-angular/src/app/models/prescription.ts
import { Doctor } from "./doctor";
import { Drug } from "./drug";
import { User } from "./user";

export interface Prescription {
  id: string;
  date: string;
  endDate: string;
  dose: number;
  userId: string;
  user?: User;
  drugId: string;
  drug?: Drug;
  doctorId: string;
  doctor?: Doctor;
}

File: MedprDB/MedprDB/Entities/Prescription.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedprDB.Entities
{
    public class Prescription : IBaseEntity
    {
        public Guid Id { get; set; }
        public string? NotificationId { get; set; }

        [Column(TypeName = "DateTime2")]
        public DateTime Date { get; set; }

        [Column(TypeName = "DateTime2")]
        public DateTime EndDate { get; set; }

        public int Dose { get; set; }

        public Guid UserId { get; set; }
        public User User { get; set; }

        public Guid DrugId { get; set; }
        public Drug Drug { get; set; }

        public Guid DoctorId { get; set; }
        public Doctor Doctor { get; set; }
    }
}